packages/ui/components/apps/AppCard.tsx

```diff
@@ -170,3 +167,100 @@ export function AppCard({ app, credentials, searchText }: AppCardProps) {
     </div>
   );
 }
+
+const InstallAppButtonChild = ({
+  userAdminTeams,
+  addAppMutationInput,
+  appCategories,
+  credentials,
+  ...props
+}: {
+  userAdminTeams?: UserAdminTeams;
+  addAppMutationInput: { type: App["type"]; variant: string; slug: string };
+  appCategories: string[];
+  credentials?: Credential[];
+} & ButtonProps) => {
+  const { t } = useLocale();
+  const router = useRouter();
+
+  const mutation = useAddAppMutation(null, {
+    onSuccess: (data) => {
+      // Refresh SSR page content without actual reload
+      router.replace(router.asPath);
+      if (data?.setupPending) return;
+      showToast(t("app_successfully_installed"), "success");
+    },
+    onError: (error) => {
+      if (error instanceof Error) showToast(error.message || t("app_could_not_be_installed"), "error");
+    },
+  });
+
+  if (
+    !userAdminTeams?.length ||
+    appCategories.some((category) => category === "calendar" || category === "video")
+  ) {
+    return (
+      <Button
+        color="secondary"
+        className="[@media(max-width:260px)]:w-full [@media(max-width:260px)]:justify-center"
+        StartIcon={Plus}
+        data-testid="install-app-button"
+        {...props}>
+        {t("install")}
+      </Button>
+    );
+  }
+
+  return (
+    <Dropdown>
+      <DropdownMenuTrigger asChild>
+        <Button
+          color="secondary"
+          className="[@media(max-width:260px)]:w-full [@media(max-width:260px)]:justify-center"
+          StartIcon={Plus}
+          data-testid="install-app-button"
+          {...props}>
+          {t("install")}
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuPortal>
+        <DropdownMenuContent>
+          <DropdownMenuLabel>{t("install_app_on")}</DropdownMenuLabel>
+          {userAdminTeams.map((team) => (
+            <DropdownItem
+              type="button"
+              disabled={
+                credentials &&
+                credentials.some((credential) =>
+                  credential?.teamId ? credential?.teamId === team.id : credential.userId === team.id
+                )
+              }
+              key={team.id}
+              StartIcon={(props) => (
+                <Avatar
+                  alt={team.logo || ""}
+                  imageSrc={team.logo || `${CAL_URL}/${team.logo}/avatar.png`} // if no image, use default avatar
+                  size="sm"
+                  {...props}
+                />
+              )}
+              onClick={() => {
+                mutation.mutate(
+                  team.isUser ? addAppMutationInput : { ...addAppMutationInput, teamId: team.id }
+                );
+              }}>
+              <p>
+                {team.name}{" "}
+                {credentials &&
+                  credentials.some((credential) =>
+                    credential?.teamId ? credential?.teamId === team.id : credential.userId === team.id
```

Comment:
It would be better to extract it out in a variable called `isInstalledForTeamOrUser` as it's used in 2 places.